icepyx/tests/test_behind_NSIDC_API_login.py

```diff
@@ -0,0 +1,47 @@
+import icepyx as ipx
+import pytest
+import warnings
+
+
+# Misc notes and needed tests
+# test avail data and subsetting success for each input type (kml, shp, list of coords, bbox)
+# check that downloaded data is subset? or is this an NSIDC level test so long as we verify the right info is submitted?
+
+
+@pytest.fixture
+def reg(scope="module"):
+    return ipx.Query("ATL06", [-55, 68, -48, 71], ["2019-02-22", "2019-02-28"])
+
+
+@pytest.fixture
+def session(reg, scope="module"):
+    return reg.Earthdata._start_session(
+        "icepyx_devteam", "icepyx.dev@gmail.com", os.getenv("NSIDC_LOGIN")
+    )
+
+
+########## is2ref module ##########
+import icepyx.core.is2ref as is2ref
+import json
+
+
+def test_get_custom_options_output(session):
+    obs = is2ref._get_custom_options(session, "ATL06", "002")
+    with open("./ATL06v04_options.json", "r") as exp:
+        assert all(keys in obs.keys() for keys in exp.keys())
+        assert all(obs[key] == exp[key] for key in exp.keys())
+
+
+########## query module ##########
+# NOTE: best this test can do at the moment is a successful download with no errors...
+def test_download_granules_with_subsetting(reg_a, session):
+    path = "./downloads_subset"
+    reg_a.order_granules(session)
+    reg_a.download_granules(session, path)
+
+
+# def test_download_granules_without_subsetting(reg_a, session):
+#     path = './downloads'
+#     reg_a.order_granules(session, subset=False)
+#     reg_a.download_granules(session, path)
+#     #check that the max extent of the downloaded granules isn't subsetted
```

Comment:
@wallinb Would you recommend that we try to do some sort of testing on our end to make sure we're getting correctly subsetted files, or is it sufficient to test that we're sending proper requests to the subsetter (I think this still needs to be implemented, but that's besides the point) and the accuracy of the input/output of the subsetter is tested on the NSIDC end?

Reply:
@wallinb Any thoughts on this? I'd like to get this PR ready for review, so that we can move forward with #200.

Reply:
I think this is still a relevant question, but there are some staff changes over at NSIDC that may delay getting an answer to this in the short term, so I'm going to resolve this for now and have added it to my long dev item list.